aerogear/mobile-security-service
# feat(get-apps-by-id): Endpoint get apps by id (AEROGEAR-8350)

## Motivation
https://issues.jboss.org/browse/AEROGEAR-8350

## What
Endpoint get apps by id 

## Why
Required by project spec

## How
Create route, handler, service and repository

## Verification Steps
- Run the server and check the following urls
```bash
go run cmd/mobile-security-service/main.go
```
Seed the database with
```sql
INSERT INTO app
			(id, app_id, app_name, deleted_at)
		VALUES 
			('1b9e7a5f-af7c-4055-b488-72f2b5f72266', 'com.aerogear.foobar', 'Foobar', NULL),
			('ae2da1f5-a9c4-4305-84bc-80da683fbc36', 'com.test.app1', 'App One', '2019-02-18 14:36:35'),
			('0890506c-3dd1-43ad-8a09-21a4111a65a6', 'com.aerogear.testapp', 'Test App', NULL);

		INSERT INTO version
			(id, version, app_id, disabled, disabled_message, num_of_app_launches)
		VALUES 
			('f6fe70a3-8c99-429c-8c77-a2efa7d0b458', '1', 'com.aerogear.testapp', FALSE, '', 5000),
    	('9bc87235-6bcb-40ab-993c-8722d86e2201', '1.1', 'com.aerogear.testapp', TRUE, 'Please contact an administrator', 1000),
			('def3c38b-5765-4041-a8e1-b2b60d58bece', '1', 'com.test.app1', FALSE, '', 10000);
				
		INSERT INTO device
			(id, version_id, app_id, device_id, device_type, device_version)
		VALUES 
			('d19feeb4-fb21-44e8-9990-473bf97a0a3f', 'f6fe70a3-8c99-429c-8c77-a2efa7d0b458', 'com.aerogear.testapp', 'a742f8b7-5e2f-43f3-a3c8-073da858420f', 'iOS', '10.2'),
			('00cb8957-db04-4ab6-8fd8-14b9fc516dbd', '9bc87235-6bcb-40ab-993c-8722d86e2201', 'com.aerogear.testapp', 'd1895cc1-28d7-4283-932d-8bcab9e4a461', 'Android', '3.2'),
			('e3b43b01-167b-48ef-8ff4-caf2e6613dee', '9bc87235-6bcb-40ab-993c-8722d86e2201', 'com.aerogear.testapp', 'feee7f81-0e33-4548-abbb-17a681c12f3b', 'Android', '4.1'),
			('ab411c3e-29f8-4e70-9ddc-8bafbba3fc4c', 'def3c38b-5765-4041-a8e1-b2b60d58bece', 'com.test.app1', '94da9833-093e-4f4c-9a93-b11600ce46b7', 'iOS', '2.0'),
			('a42a128a-dfb6-435c-8653-8f66ab3a5a1c', 'def3c38b-5765-4041-a8e1-b2b60d58bece', 'com.test.app1', '94132b0c-d7b1-4419-bcce-fc6760c59e3a', 'Android', '4.1');
```

- invalid id requests like 
http://localhost:3000/api/apps/a986f8d1-aff7-45e0-8fc6-cec2abc0234d;DROP TABLE
will return 
```js
{
  message: "Invalid id supplied",
  statusCode: 400
}
```
- id not found error 
http://localhost:3000/api/apps/d986f8d1-aff7-45e0-8fc6-cec2abc0234d
will return
```js
{
  message: "App not found",
  statusCode: 404
}
```
- Successful request
http://localhost:3000/api/apps/0890506c-3dd1-43ad-8a09-21a4111a65a6
will return
```js
{
id: "0890506c-3dd1-43ad-8a09-21a4111a65a6",
appId: "com.aerogear.testapp",
appName: "Test App",
numOfCurrentInstalls: null,
numOfAppLaunches: null,
deployedVersions: [
  {
    id: "9bc87235-6bcb-40ab-993c-8722d86e2201",
    version: "1.1",
    appId: "com.aerogear.testapp",
    disabled: true,
    disabledMessage: "Please contact an administrator",
    numOfCurrentInstalls: 1000,
    numOfAppLaunches: 2
  },
  {
    id: "f6fe70a3-8c99-429c-8c77-a2efa7d0b458",
    version: "1",
    appId: "com.aerogear.testapp",
    disabled: false,
    numOfCurrentInstalls: 5000,
    numOfAppLaunches: 1
  }
]
}
```
## Checklist:


## Progress


## Additional Notes

PS.: Add images and/or .gifs to illustrate what was changed if this pull request modifies the appearance/output of something presented to the users.
